etl/steps/data/garden/un/2024-10-21/census_dates.py

```diff
@@ -0,0 +1,194 @@
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(
+        dest_dir, tables=[tb, tb_census], check_variables_metadata=True, default_metadata=ds_meadow.metadata
+    )
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+def get_comment(date: str, country: str) -> str:
+    """Get comments from footnotes in date and country column."""
+    comment = ""
+    if date.startswith("("):
+        note = date.split(" ")[0]
+        comment += COMMENTS_DICT[note]
+    if country.endswith(")"):
+        if country.split(" ")[-1] not in ["(Malvinas)"]:
+            note = country.split(" ")[-1]
+            if len(comment) == 0:
+                comment += COMMENTS_DICT[note]
+            else:
+                comment += ", " + COMMENTS_DICT[note]
+    return comment
+
+
+def clean_date(date: str) -> str:
+    """Clean date column."""
+    if date.startswith("("):
+        return " ".join(date.split(" ")[1:])
+    else:
+        return date
+
+
+def clean_country(country: str) -> str:
+    """Clean country column."""
+    if (country.endswith(")")) and (country.split(" ")[-1] not in ["(Malvinas)"]):
+        return " ".join(country.split(" ")[:-1])
+    else:
+        return country
+
+
+def date_as_year(date: str) -> int:
+    """Extract year from date."""
+    if " " in date and date.split(" ")[-1].isdigit():
+        return int(date.split(" ")[-1])
+    elif "." in date:
+        return int(date.split(".")[-1])
+    elif date.startswith("[") and date.endswith("]"):
+        return int(date[1:-1])
+    else:
+        return int(date)
+
+
+def years_since_last_census(tb: Table) -> Table:
+    countries = tb["country"].unique()
+    years = [int(x) for x in range(1985, 2024)]
+    rows = []
+    for country in countries:
+        country_df = tb[tb["country"] == country].sort_values("date_as_year", ascending=True)
+        census_years = country_df["date_as_year"].tolist()
+        for year in years:
+            prev_census = [x for x in census_years if x <= year]
+            if prev_census:
+                last_census = max([x for x in census_years if x <= year])
+                years_since_last_census = year - last_census
+            else:
+                last_census = None
+                years_since_last_census = None
+            rows.append(
+                {
+                    "country": country,
+                    "year": year,
+                    "last_census": last_census,
+                    "years_since_last_census": years_since_last_census,
+                }
+            )
+    tb_census = Table(pd.DataFrame(rows)).copy_metadata(tb)
+    tb_census.m.short_name = "years_since_last_census"
+
+    for col in tb_census.columns:
+        tb_census[col].metadata = tb["date"].m
+        tb_census[col].metadata.origins = tb["date"].m.origins
+    return tb_census
+
+
+def add_uk(tb_census):
+    years = [int(x) for x in range(1985, 2024)]
```

Comment:
you could define the years at the top so it's easier to edit on the next update. or just do max and min if appropriate

Reply:
Good point, will do!